scene/2d/tile_map.cpp

```diff
@@ -326,7 +326,8 @@ void TileMap::update_dirty_quadrants() {
 	Color debug_collision_color;
 	Color debug_navigation_color;
 
-	bool debug_shapes = st && st->is_debugging_collisions_hint();
+	bool debug_shapes = show_collision && (Engine::get_singleton()->is_editor_hint() || st && st->is_debugging_collisions_hint());
```

Comment:
Actually it needs extra parenthesis to fix compilation errors on some platforms:
`bool debug_shapes = show_collision && (Engine::get_singleton()->is_editor_hint() || (st && st->is_debugging_collisions_hint()));`